lib/validate.js

```diff
@@ -0,0 +1,29 @@
+module.exports = validate
+
+var Promise = require('lie')
+
+function validate (state, doc) {
+  var self = this
```

Comment:
If we would leave state.validate unset if the options wasn’t set in the constructor, then we could check if it is set on this line and if it’s not directly `return Promise.resolve()`